### PR TITLE
Bump redux and fix CI

### PR DESCRIPTION
**Pull request by reflog**
#### Summary
Bump redux to the typescript one
Fix CI process (removed redundant step)

### Related PR
https://github.com/mattermost/mattermost-server/pull/12937

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -28,7 +28,6 @@ jobs:
       - *restore_cache
       - run: |
           npm ci
-          cd node_modules/mattermost-redux && npm i && npm run build
```
Comment (hanzei) [inline .circleci/config.yml]: Can you please elaborate why we don't want to cache the `node_modules` from redux any longer?

Comment (reflog) [reply .circleci/config.yml]: This step is no longer needed for building webapp. Not related to cache

**Comment by reflog** [comment]
> I don't get what this PR fixes

New redux doesn't need the extra 'npm install' from inside node_modules, so this step was removed

**Comment by jespino** [comment]
@hmhealey PTAL

**Comment by jespino** [comment]
@hanzei PTAL

**Comment by hanzei** [comment]
@jespino please request a review from someone else. I don't have enough domain knowledge to review this.

**Comment by reflog** [comment]
@jespino - you rule. Thanks for fixing my mess 🤣

**Comment by jespino** [comment]
@hmhealey we could try to rethink how we do it. Clearly is not the best approach possible because, as you say, the behavior when you use git is not the expected behavior.

One option is to always publish to npm every single commit (not super sexy, but can work), other option is to redesign mattermost-redux to allow us to use it directly from source code as typescript, instead of compiling it on install.

**Comment by hmhealey** [comment]
I'd love to be able to include mattermost-redux directly into the apps without compiling it, but we need to figure out how to make that work with source maps and everything. I forget why exactly we chose to compile it

**Comment by reflog** [comment]
@jespino , @hmhealey - we can easily combine the two approaches:
All projects that already use mattermost-redux as we do right now can continue to do so
Webapp can include mattermost-redux as git submodule (easier to point to specific hashes) and will be built by webpack+typescript. The only thing to do is to add 'alias' to webpack config and we should be fine!
If there's interest - I'll be more than glad to experiment on this.

**Pull request merged**